src/python/pants/jvm/dependency_inference/artifact_mapper.py

```diff
@@ -279,5 +280,68 @@ def insert(
     )
 
 
+class ConflictingJvmArtifactVersion(ValueError):
+    def __init__(
+        self, group: str, artifact: str, required_version: str, found_coordinate: Coordinate
+    ) -> None:
+        self.group: str = group
+        self.artifact: str = artifact
+        self.required_version: str = required_version
+        self.found_coordinate: Coordinate = found_coordinate
+
+
+class MissingJvmArtifacts(ValueError):
+    def __init__(self, coordinates: Iterable[Coordinate | UnversionedCoordinate]) -> None:
+        self.coordinates: tuple[Coordinate | UnversionedCoordinate, ...] = tuple(coordinates)
+
+
+def find_jvm_artifacts_or_raise(
+    required_coordinates: Iterable[Coordinate | UnversionedCoordinate],
```

Comment:
This takes multiple coordinates because the Scrooge Scala backend will need to check for multiple coordinates.